### DISCUSSION TITLE
Config entry for each Jira instance and its credentails

### DISCUSSION THREAD

**Discussion by electriquo** [ankitpokhrel/jira-cli#435] [Ideas]
**Thank you for the JiraCLI.**

JiraCLI requires to [expose Jira API Token (or password)](https://github.com/ankitpokhrel/jira-cli/blob/3d15182e686f62fa56b50bd5a78ea9e30cce6f71/README.md?plain=1#L116) and maybe [tweaking the authentication type](https://github.com/ankitpokhrel/jira-cli/blob/3d15182e686f62fa56b50bd5a78ea9e30cce6f71/README.md?plain=1#L144) for Jira CLI to work. One might even consider other means of [exposing Jira API Token (or password)](https://github.com/ankitpokhrel/jira-cli/discussions/356).

The process above needs to be repeated for every Jira instance.
After the process was done, the modified files (e.g. `.bashrc`, `.netrc`, etc.) should not be stored under Source Control Management (SCM) such as GIt due to the sensitivity of the credentials in these files.

What about letting `jira init` to open the internet browser and point it to [Jira API Tokens](https://id.atlassian.com/manage-profile/security/api-tokens), then wait for the user to type in the API token once created. In turn JiraCLI will create a file `$HOME/.config/.jira/.config.yml` with all the configuration and the credentials in the same [spirit GitHub CLI](https://github.com/ankitpokhrel/jira-cli/blob/3d15182e686f62fa56b50bd5a78ea9e30cce6f71/README.md?plain=1#L48) keeps such [configuration and credentials](https://github.com/cli/cli/blob/d21d388b8dc10c8f04187c3afa6e0b44f0977c65/internal/config/config.go#L18-L28)?

This will:
1. Allow to exclude the file from SCM tracking and without compromising any credentials leakage into the profile file, which cannot be excluded from the SCM for obvious reasons
2. Streamline the authentication processes by removing the requirement to (a) read all the instructions on how to set the token, (b) modifying any files manually, especially will discard the modification of broad user files, and (c) easing the management of such many configurations using `--config`.

related to #356, #434, #268, #217, #287 (and maybe some others).

**Comment by ankitpokhrel** [comment]
Hi @foolioo, thank you for opening this thread. I think this is a valid concern and something that could be improved (I am open for PRs). 

Having said that, there are a ton of things that GitHub CLI does better and can be borrowed to make JiraCLI better. But, right now its just a single person maintaining this project and I am trying my best to make the tool usable for majority of use cases. Any kind of [support](https://github.com/ankitpokhrel/jira-cli#support-the-project) is highly appreciated.

**Reply by electriquo** [reply]
**Thank you for all your efforts!**

I would really love to contribute, but I won't be of any help as I am not fluent in Go.
I've included a link to GitHub CLI implementation, maybe it will give a lead to one who can contribute to the project.

Right now, this thing kinda blocks me from adopting Jira CLI and enjoy your work.
Do you think we should convert this discussion into a feature request?

**Reply by electriquo** [reply]
@ankitpokhrel Do you think we should convert this discussion into a feature request?
